2020/4/passport/src/main.rs

```diff
@@ -0,0 +1,248 @@
+use std::env;
+use std::fs::File;
+use std::io::{self, BufRead};
+
+fn main() {
+    let args: Vec<String> = env::args().collect();
+    let file = File::open(&args[1]);
+
+    let reader = io::BufReader::new(file.unwrap());
+    let mut current = String::from("");
+    let mut passports = Vec::new();
+    for line in reader.lines() {
+        let l = line.unwrap();
+        if l.len() == 0 {
+            let passport = Passport::from_string(&current);
```

Comment:
In "productiony" code I'd expect `from_string` to return a `Result<Passport, Error>` of some kind. Might be interesting for you to explore error handling in Rust. What I usually do is have an `Error` enum in my crate, something like:

```rust
#[derive(Debug, Eq, PartialEq)]
pub enum Error {
    InvalidPassport,
    UnknownFieldName(String),
    // ... other named types of errors that can occur ...
}
```

Not at all "necessary" for AOC, just something you could play with.